Documenting ways for in-file and cross-file navigation (#32154)
quarkusio/quarkus: docs/src/main/asciidoc/doc-contribute-docs-howto.adoc

For more information about the minimum header requirements, see xref:{doc-guides}/doc-reference.adoc#document-structure[Document structure] on the "Quarkus style and content guidelines" page.

[[anchors-howto]]

> === Cross-referencing in-file and cross-file content by using anchors

Reviewer: Just a very minor suggestion: Use the imperative verb form instead of the gerund form, as per the upstream Quarkus style preference for titles. 

```suggestion
=== Cross-reference in-file and cross-file content by using anchors
```

Author: Done and done

[source,asciidoc]
----
[[title-heading]]
== Title heading
----

==== Call an anchored ID in the same file

To call an anchor created in the same file, insert the anchored ID in a `<<>>` xref macro.

.Inter-document anchored ID call example
[source,asciidoc]
----
<<title-heading>>
----

This macro creates an URL with a name automatically sourced from the anchored heading, section, or table.

WARNING: Do not use the `<<>>` format with the verbatim heading or section description, such as
`\<<Title heading>>`.

When you want to specify a non-default caption for your URL, specify the anchored ID and desired name separated by `,` without white space.

.Anchor with a custom URL caption example
[source,asciidoc]
----
<<title-heading,Title heading description that fits the context of your content>>
----

==== Call an anchored ID from a different file

To call an anchor created in a different file, insert the anchor to an `xref` macro and specify the full name of the hosting file and the desired anchored ID.

.Cross-document anchored ID call example
[source,asciidoc]
----
xref:<other-file-name>.adoc#title-heading[Title heading]
----

> Thus, use the `xref:<name-of-the-file>.adoc[Human-readable label]` syntax to refer to another file.

Reviewer 2: ```suggestion
Breaking this example apart, we are using the `xref` macro to refer to another file (`xref:<name-of-the-file>.adoc[Human-readable label]`) and inserting the anchor ID for the target section (`#title-heading`) just after the file name. 
```

> See also the xref:doc-reference.adoc#cross-references[Cross-references] section, which provides information about the use of the `/path` attributes.

Reviewer 2: ```suggestion
See the xref:doc-reference.adoc#cross-references[Cross-references] section for path attributes that should be used in cross-document references.
```
